### PR TITLE
add methods to schedules model & api

### PR DESCRIPTION
**Pull request by BetWithFriend**
Check the new functions.
Also try to make quick fix for the register.

### CODE REVIEW

File: leagues/api.py [hunk 1]
```diff
@@ -78,3 +78,26 @@ def group_get(user, qs):
 
     return result
 
+
+@jsonrpc.method_save('schedule.save(id=Number, fields=Object, group_id=Number)', model=Schedule)
+def schedule_save(user, obj, group_id):
+    if not group_id:
+        raise GeneralError("group_id field is mandatory")
+    else:
+        obj.league_id = group_id
```
Comment (shlimp) [inline leagues/api.py]: it should be obj.group_id = group_id, not obj.league_id

File: leagues/api.py [hunk 2]
```diff
@@ -78,3 +78,26 @@ def group_get(user, qs):
...
+    if not obj.is_owner(user):
+        raise PermissionError("save", "Schedule")
+
+    db.session.add(obj)
+    db.session.commit()
+    return obj.asdict(only=["id", "home_team_id", "away_team_id"], follow={"group": {"only": ["id", "name"]}})
```
Comment (shlimp) [inline leagues/api.py]: instead of returning just the team ids, return a home_team and away_team objects using follow

File: leagues/api.py [hunk 3]
```diff
@@ -78,3 +78,26 @@ def group_get(user, qs):
...
+
+
+@jsonrpc.method_get('schedule.get(filter=Object)', model=Schedule)
+def schedule_get(user, qs):
+    result = []
+    for item in qs.all():
+        result.append(
+            item.asdict(only=["id", "home_team_id", "away_team_id"], follow={"group": {"only": ["id", "name"]}}))
```
Comment (shlimp) [inline leagues/api.py]: same as above comment, return objects

File: leagues/models.py [hunk 4]
```diff
@@ -95,6 +95,20 @@ class Schedule(db.Model, DictableModel):
     group_id = db.Column(db.Integer, db.ForeignKey('group.id'))
     group = db.relationship('Group', backref=db.backref('schedules', lazy='dynamic'))
 
+    def is_owner(self, request_user):
+        return request_user.id == self.group.league.owner_id
+
+    def can_add(self, request_user, group_id, **kwargs):
+        group = Group.query.get(group_id)
+        return request_user.id == group.league.owner_id
```
Comment (shlimp) [inline leagues/models.py]: use return self.is_owner instead. no point in writing the same like twice

**Comment by BetWithFriend** [comment]
i've fixed all your comments.
Still need to check the return value of save & get.
please fix the register and add logout.. :)

**Pull request merged**